saleor/payment/utils.py

```diff
@@ -1,38 +1,178 @@
 import json
 import logging
+from collections import defaultdict
 from decimal import Decimal
-from typing import TYPE_CHECKING, Dict, List, Optional
+from itertools import chain
+from typing import Any, Dict, Iterator, List, Optional, Tuple
 
 import graphene
 from babel.numbers import get_currency_precision
 from django.core.serializers.json import DjangoJSONEncoder
 
 from ..account.models import User
+from ..checkout.calculations import checkout_line_total
+from ..checkout.fetch import fetch_checkout_info, fetch_checkout_lines
 from ..checkout.models import Checkout
 from ..core.prices import quantize_price
 from ..core.tracing import traced_atomic_transaction
-from ..order.models import Order
+from ..discount.utils import fetch_active_discounts
+from ..order import FulfillmentLineData, FulfillmentStatus, OrderLineData
+from ..order.models import FulfillmentLine, Order, OrderLine
+from ..plugins.manager import PluginsManager, get_plugins_manager
 from . import ChargeStatus, GatewayError, PaymentError, TransactionKind
 from .error_codes import PaymentErrorCode
-from .interface import AddressData, GatewayResponse, PaymentData, PaymentMethodInfo
+from .interface import (
+    AddressData,
+    GatewayResponse,
+    PaymentData,
+    PaymentLineData,
+    PaymentMethodInfo,
+)
 from .models import Payment, Transaction
 
-if TYPE_CHECKING:
-    from ..plugins.manager import PluginsManager
-
 logger = logging.getLogger(__name__)
 
 GENERIC_TRANSACTION_ERROR = "Transaction was unsuccessful"
 ALLOWED_GATEWAY_KINDS = {choices[0] for choices in TransactionKind.CHOICES}
 
 
+def create_payment_lines_information(
+    payment: Payment,
+    manager: PluginsManager,
+) -> List[PaymentLineData]:
+    checkout = payment.checkout
+    order = payment.order
+    line_items = []
+    if checkout:
```

Comment:
Minor thing but this function is pretty big, it would be a bit more readable if we added some separate helpers for the checkout case and the order case below.